server/src/main/java/io/crate/replication/logical/SynchronizeTableDefinitionsTask.java

```diff
@@ -0,0 +1,212 @@
+import org.elasticsearch.action.admin.cluster.state.ClusterStateAction;
+import org.elasticsearch.action.admin.cluster.state.ClusterStateResponse;
+import org.elasticsearch.action.bulk.BackoffPolicy;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.ClusterStateUpdateTask;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.logging.Loggers;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.concurrent.EsRejectedExecutionException;
+import org.elasticsearch.threadpool.Scheduler;
+import org.elasticsearch.threadpool.ThreadPool;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.function.Function;
+
+import static io.crate.replication.logical.repository.LogicalReplicationRepository.REMOTE_CLUSTER_REPO_REQ_TIMEOUT_IN_MILLI_SEC;
+
+public final class SynchronizeTableDefinitionsTask implements Closeable {
+
+    private static final Logger LOGGER = Loggers.getLogger(SynchronizeTableDefinitionsTask.class);
+
+    private final ThreadPool threadPool;
+    private final Function<String, Client> remoteClient;
+    private final ClusterService clusterService;
+    private final Set<String> subscriptionsToTrack = Collections.synchronizedSet(new HashSet<>());
+    private final TimeValue pollDelay;
+    private final Iterator<TimeValue> delay;
+
+    private Scheduler.Cancellable cancellable;
+    private boolean isStarted = false;
+
+    public SynchronizeTableDefinitionsTask(Settings settings, ThreadPool threadPool, Function<String, Client> remoteClient, ClusterService clusterService) {
+        this.threadPool = threadPool;
+        this.remoteClient = remoteClient;
+        this.clusterService = clusterService;
+        this.pollDelay = LogicalReplicationSettings.REPLICATION_READ_POLL_DURATION.get(settings);
+        this.delay = BackoffPolicy.exponentialBackoff(pollDelay, 8).iterator();
+    }
+
+    public void start() {
+        assert isStarted == false : "SynchronizeTableDefinitionsTask is already started";
+        assert clusterService.state().getNodes().getLocalNode().isMasterNode() : "SynchronizeTableDefinitionsTask must only be executed on the master node";
+        var executor = threadPool.executor(ThreadPool.Names.LOGICAL_REPLICATION);
+
+        try {
+            executor.execute(this::run);
+        } catch (EsRejectedExecutionException e) {
+            threadPool.schedule(
+                this::run,
+                delay.next(),
+                ThreadPool.Names.LOGICAL_REPLICATION
+            );
+        }
+        isStarted = true;
+    }
+
+    public boolean addRemoteClusterToTrack(String remoteCluster) {
```

Comment:
How and when are subscriptions removed?

Reply:
I added the method for this, and we need to use them once we implement the logic for `DROP SUBSCRIPTION`.